### PR TITLE
Add minimal shell test suite and allow testing against docker hawkbit instance

### PR DESCRIPTION
**Pull request by ejoerns**
This series consists of mutliple pre-work and adaptions required for running a minimum set of runtime tests for rauc-hawkbit-updater.

* Add sharness shell test suite
* Prepare code for returning error codes in ase of failures
* spawn hawkbit using docker from travis test
* simple tests for token-based authentification with hawkbit

### CODE REVIEW

File: src/rauc-hawkbit-updater.c [hunk 1]
```diff
@@ -118,6 +118,12 @@ int main(int argc, char **argv)
                 goto out;
         }
 
+        if (config_file == NULL) {
+                g_printerr("No configuration file given\n");
+                exit_code = 3;
```
Comment (Raphexion) [inline src/rauc-hawkbit-updater.c]: Would it be better to change the error_code to something different than 3? Three is already used below (No such configuration file).

**Comment by Raphexion** [comment]
I think it is a good pull request. Thank you for the well written commit messages.

LGTM. @prevas-lkmi has the last say.

File: src/rauc-hawkbit-updater.c (hunk 1 above)
Comment (ejoerns) [reply src/rauc-hawkbit-updater.c]: I also thought about that, but it was not clear to me what the error code says and I interpreted it as 'configuration file issue'. We could also return 4 here, but maybe we need a better definition of error return codes, anyway. I do not have a real strong opinion on this, yet. What would you prefer?

Comment (Raphexion) [reply src/rauc-hawkbit-updater.c]: I don't have a strong opinion. Imagine that we change it to 4. Could we then create an extra test case? I am thinking that we can have two different errors:
1. No configure file provided (null).
2. Config file provided but it does not exist.

Comment (ejoerns) [reply src/rauc-hawkbit-updater.c]: I've now changed the exit code and also added three more tests covering the cases a) invalid argument, b) missing config argument, c) path to non-existing config

Comment (Raphexion) [reply src/rauc-hawkbit-updater.c]: Great!

**Pull request merged**